Sources/SwiftDocC/Semantics/MarkupReferenceResolver.swift

```diff
@@ -112,24 +118,19 @@ struct MarkupReferenceResolver: MarkupRewriter {
         return link
     }
 
-    mutating func resolveAbsoluteSymbolLink(unresolvedDestination: String, elementRange range: SourceRange?) -> String {
+    mutating func resolveAbsoluteSymbolLink(unresolvedDestination: String, elementRange range: SourceRange?) -> ResolvedTopicReference? {
         if let cached = context.documentationCacheBasedLinkResolver.referenceFor(absoluteSymbolPath: unresolvedDestination, parent: rootReference) {
             guard context.topicGraph.isLinkable(cached) == true else {
                 problems.append(disabledLinkDestinationProblem(reference: cached, source: source, range: range, severity: .warning))
-                return unresolvedDestination
+                return nil
```

Comment:
If I understand correctly, this change doesn't change the behavior, because the previous was essentially setting `symbolLink.destination` to the same value it contained before?

Reply:
Correct, no functional change to the resolution logic. Just makes it easier to use the resolved reference in the snippet case.